src/com/directi/training/codesmells/smelly/Position.java

```diff
@@ -0,0 +1,27 @@
+package com.directi.training.codesmells.smelly;
+
+public class Position
+{
+    public final int row, column;
```

Comment:
The general pattern followed for pojos is to have private fields and expose getters and setters if required. So we should revert the change. I had not followed this in the Pair class because I was just mimicking how the sun provided pair class was operating but yes we can make the fields private in that class also and expose getters.